### PR TITLE
TweetTopicSingleClassification added

### PR DESCRIPTION
**Pull request by dokato**
## Checklist for adding MMTEB dataset

Reason for dataset addition:

Well structured: https://huggingface.co/datasets/cardiffnlp/tweet_topic_single and described: https://arxiv.org/abs/2209.09824 dataset. Easy gain.

- [x] I have tested that the dataset runs with the `mteb` package.
- [x] I have run the following models on the task (adding the results to the pr). These can be run using the `mteb run -m {model_name} -t {task_name}` command.
  - [x] `sentence-transformers/paraphrase-multilingual-MiniLM-L12-v2`
  - [x] `intfloat/multilingual-e5-small`
- [x] I have checked that the performance is neither trivial (both models gain close to perfect scores) nor random (both models gain close to random scores).
- [x] If the dataset is too big (e.g. >2048 examples), considering using `self.stratified_subsampling() under dataset_transform()`
- [x] I have filled out the metadata object in the dataset file (find documentation on it [here](https://github.com/embeddings-benchmark/mteb/blob/main/docs/adding_a_dataset.md#2-creating-the-metadata-object)).
- [x] Run tests locally to make sure nothing is broken using `make test`. 
- [x] Run the formatter to format the code using `make lint`. 
- [ ] I have added points for my submission to the [points folder](https://github.com/embeddings-benchmark/mteb/blob/main/docs/mmteb/points.md) using the PR number as the filename (e.g. `438.jsonl`).

### CODE REVIEW

File: mteb/tasks/Classification/eng/TweetTopicSingleClassification.py [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+        },
+        reference="https://arxiv.org/abs/2209.09824",
+        type="Classification",
+        category="s2s",
+        eval_splits=["test_2021"],
+        eval_langs=["eng-Latn"],
+        main_score="accuracy",
+        date=("2019-09-01", "2021-08-31"),
+        form=["written"],
+        domains=["News"],
```
Comment (KennethEnevoldsen) [inline mteb/tasks/Classification/eng/TweetTopicSingleClassification.py:28]:
```suggestion
        domains=["Social", "News"],
```

File: mteb/tasks/Classification/eng/TweetTopicSingleClassification.py [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+            publisher = "International Committee on Computational Linguistics"
+        }
+        """,
+        n_samples={"test_2021": 1693},
+        avg_character_length={"test_2021": 167.66},
+    )
+
+    def dataset_transform(self):
+        self.dataset["train"] = self.dataset["train_2021"]
+        return super().dataset_transform()
```
Comment (KennethEnevoldsen) [inline mteb/tasks/Classification/eng/TweetTopicSingleClassification.py:57]:
```suggestion
```

**Review by KennethEnevoldsen — APPROVED** [review]
Looks good feel free to add points as well. You could also consider reformulating it as a clustering task (it might be better suited for it).

**Comment by dokato** [comment]
Thanks a lot for review @KennethEnevoldsen ! I decided to keep it classification for now, as I keep my eye at the new MultiLabel task to be added, and then I'd like to add also multilabel version of this (https://huggingface.co/datasets/cardiffnlp/tweet_topic_multi) which may confuse people, why one is clustering and the other classification, but if you feel strongly about it can move, no problem.

**Review by KennethEnevoldsen — APPROVED** [review]

**Pull request merged**